transport-native-unix-common/src/main/c/netty_unix_socket.c

```diff
@@ -1001,27 +997,24 @@ jint netty_unix_socket_JNI_OnLoad(JNIEnv* env, const char* packagePrefix) {
         netty_unix_errors_throwRuntimeException(env, "failed to get method ID: InetSocketAddress.<init>(String, int)");
         return JNI_ERR;
     }
-    nettyClassName = netty_unix_util_prepend(packagePrefix, "io/netty/util/NetUtil");
-    jclass localNetUtilClass = (*env)->FindClass(env, nettyClassName);
-    free(nettyClassName);
-    nettyClassName = NULL;
-    if (localNetUtilClass == NULL) {
+
+    // We retrieve java.net.preferIPv4Stack directly via Boolean.getBoolean(...) and not via NetUtil to ensure
+    // we not run into deadlocks while trying to load classes that may depend on each other.
+    // See https://github.com/netty/netty/issues/7458
+    jclass booleanClass = (*env)->FindClass(env, "java/lang/Boolean");
```

Comment:
This is really hacky.  IMO the better long term solution would be to pass in all the dependencies (like isipv4Preferred) using some `Java_io_netty_epoll_init()` like method.  

As is it's probably fine.

Reply:
Sure I can try this as well just thought this is good enough as it’s the same as openjdk does without a problem... WDYT?

Reply:
OpenJDK is in a unique position because it also controls all the classes that are being loaded at the time.  The JDK can take shortcuts that in general outside code can't.  (side note: this is also true for concurrent classes!).  


Like I said this is fine as is for working around the current issue.   However, if any other native loading happens in any of the other JNI_OnLoad  methods they would need to be fixed too.  My concern was that it will be long term brittle.

Reply:
@carl-mastrangelo got it... let me try to find time to come-up with another pr.

Reply:
+1 for passing info into JNI if possible instead of JNI reaching back to java.